coldfront/plugins/qumulo/utils/acl_allocations.py

```diff
@@ -142,6 +145,14 @@ def get_access_allocations(qumulo_allocation: Allocation) -> list[Allocation]:
 
         return list(access_allocations)
 
+    @staticmethod
+    def is_base_allocation(path: str):
```

Comment:
@harterjwustl I thought the sub-allocation should try finding its parent before using the path matching. I would least leave a comment indicating the design decision for future development.

Reply:
The relationship between the sub-allocation and its parent through the allocation linkage is robust and enduring. Making the relationship through the path is a weaker reference. As long as the paths are immutable, we should be okay. Nonetheless, the Allocation Linkage reference should eventually be the way to go.

Reply:
So, as mentioned, I made a ticket for this already.  I really prefer not to leave information like that in the middle of the codebase.  It just ends up as comment garbage the never gets taken out, in my experience.

Path's are immutable, but I agree about the Allocation Linkage.  Unfortunately at the moment there is not enough safeguards in place and too much nonconforming data for the Allocation Linkage to be reliable.  As mentioned in app eng, we have moved this to it's own ticket to prevent scope-creep on what is a bug fix.  It's something we have talked about recently and I think this is a good place to set an example and try to improve our cadence.

An aside: I am not sure what you mean about a sub-allocation finding its parent or how that is relevant in this method.  We are just trying to determine if an allocation is a base/root/parent.  In the future it could look something like:

```python
def is_base_allocation(allocation: Allocation):
    allocationLinkage = AllocationLinkage.objects.filter(children__contains(allocation))
    
    return len(AllocationLinkage) == 0
```

Reply:
This issue should be revisited in a design meeting. Leave it as implemented.